### DISCUSSION TITLE
Callback for UAP0 up/down events

### DISCUSSION THREAD

**Discussion by FIAV1** [RaspAP/raspap-webgui#892] [Q&A]
Is there a way to use a callback for UAP0 up/down events? I never did this before, I tried to place a script in `/etc/network/if-up.d` and `/etc/network/if-down.d` but they don't work. Any hints?

**Comment by billz** [comment]
The scripts in `/etc/network/if-*.d` are executed by ifupdown in pre-up, post-up, pre-down and post-down phases of network interface configuration. This is the pre-Debian Jessie paradigm which has now been superseded by dhcpcd. 

Have you looked at `ip monitor`? This is a low level tool that could be used to respond to these events. The uap0 interface is created with iw and ifconfig, which are higher level tools. A callback might take the form:

`ip monitor address dev uap0 | while read event; do ...`
